pkg/crc/logging/logging.go

```diff
@@ -18,6 +18,10 @@ var (
 )
 
 func OpenLogFile() (*os.File, error) {
+	err := constants.EnsureBaseDirExists()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
@cfergeau This should be part of `crc-embedder/root.go` script instead here, same way it is part of `crc/root.go` because it is not only creating the log directory but initial structure so should be part of init method of root.

Reply:
With Anjan's refactor of this code, we can move the responsibility of creating the directories to the caller.
With the current code, where it's `pkg/crc/logging/logging.go` which decides where to put its log file, to me it makes sense that it also makes sure the directory exists.

Reply:
@cfergeau you mean https://github.com/code-ready/crc/pull/861/commits/1f9070d6146b63f2a443c03c6433146022bd8012 this refactor? This is just adding a path argument, or it is a different one?

Reply:
Yes, this one. Currently, where the log is stored is an implementation detail of the 'logging' package, so in my opinion it makes sense that the `logging` package takes care of creating the path.
After Anjan's refactor, where the log is stored is specified by users of the package iirc. The location of the log file is no longer internal to the `logging` package, so it makes sense that users of `logging` are required to make sure the path exists.